### ISSUE TITLE
Binary Crossentropy give negative loss values.

### ISSUE THREAD

**Issue by PieroCV** [zzh8829/yolov3-tf2#91]
Hi!
I was trying this repo and it's amazing for pretrained weights!...
But, when i tried to train a custom model, i had some problems. First, I get NAN values :(, i changed Sparse Categorical Crossentropy for Binary Crossentropy (I'm training one class), and now I'm getting negative loss values...
My solution (not mine really) was add an extra parameter to binary crossentropy in class value:

**from_logist=True**
`class_loss = obj_mask * binary_crossentropy(
            true_class_idx, pred_class,from_logits=True)`

Now, i'm getting acceptable results, so I wanted to know if someone had the same problem.

**Comment by Yeah-Kun** [comment]
I have the same problem.

**Comment by PieroCV** [comment]
> I have the same problem.

Did you solve it? 🤔

**Comment by zzh8829** [comment]
check out this tutorial https://github.com/zzh8829/yolov3-tf2/blob/master/docs/training_voc.md first
if you have further question please reopen a new issue

**Comment by VeeranjaneyuluToka** [comment]
class_loss = obj_mask * binary_crossentropy( true_class_idx, pred_class,from_logits=True)

If i use binary corssentropy with logits true instead of sparse_categorical cross entropy, i am getting error and i have 5 classes. I know the reason as well (we get predictions as 5 classes but we have one value in gt), but wondering we need to use binary cross entropy in case if we have 1 class and categorical cross entropy for multiple classes?
